firestore/src/AggregateQuery.cs

```diff
@@ -0,0 +1,36 @@
+
+
+using Firebase.Firestore.Internal;
+using Firebase.Platform;
+using System.Threading.Tasks;
+
+namespace Firebase.Firestore {
+    public sealed class AggregateQuery
+    {
+        private readonly AggregateQueryProxy _proxy;
+        private readonly FirebaseFirestore _firestore;
+
+        internal AggregateQuery(AggregateQueryProxy proxy, FirebaseFirestore firestore) {
+            _proxy = Util.NotNull(proxy);
+            _firestore = Util.NotNull(firestore);
+        }
+
+        /// <summary>
+        /// The query of aggregations that will be calculated.
+        /// </summary>
+        public Query Query => new Query(_proxy.query(), _firestore);
+
+        public Task<AggregateQuerySnapshot> GetSnapshotAsync(AggregateSource source) {
+            var sourceProxy = Enums.Convert(source);
+            return Util.MapResult(_proxy.GetAsync(sourceProxy),
+                taskResult => { return new AggregateQuerySnapshot(taskResult, _firestore); });
+        }
+
+        public override bool Equals(object obj) => Equals(obj as Query);
```

Comment:
nit: Add the `inheritdoc` comments to the Equals and GetHashCode methods, like in AggregateQuerySnapshot.cs.